plan/planbuilder.go

```diff
@@ -599,6 +601,12 @@ func (b *planBuilder) buildShow(show *ast.ShowStmt) Plan {
 	for i, col := range p.schema.Columns {
 		col.Position = i
 	}
+	if show.Tp == ast.ShowVariables {
```

Comment:
why not handle this in line 582?

Reply:
This action should be after schema building.
So it is put after line601~604.